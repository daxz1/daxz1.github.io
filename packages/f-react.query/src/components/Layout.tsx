import { Link, Button } from "@material-ui/core";
import { Switch, Route, Link as RouterLink } from "react-router-dom";
import { makeStyles } from "@material-ui/core/styles";
import Episodes from "./Episodes";
import Episode from "./Episode";

const useStyles = makeStyles(theme => ({
    main: {
        margin: "0 auto",
        padding: "16px"
    },
    nav: {
        margin: "0 auto",
        display: "flex",
        justifyContent: "center",
        backgroundColor: "#000",
        color: "#fff",
        "& button": {
            margin: theme.spacing(1)
        }
    }
}));

export default function Layout():JSX.Element {
    const classes = useStyles();

    return (
        <div className="app">
            <nav className={classes.nav}>
                <Link component={RouterLink} to="/">
                    <Button color='primary'>Home</Button>
                </Link>
                <Link component={RouterLink} to="/episodes">
                    <Button color='primary'>Episodes</Button>
                </Link>
                <Link component={RouterLink} to="/characters">
                    <Button color='primary'>Characters</Button>
                </Link>
            </nav>
            <main className={classes.main}>
                <Switch>
                    <Route exact path="/episodes">
                        <Episodes />
                    </Route>
                    <Route exact path="/episodes/:id">
                        <Episode />
                    </Route>
                    <Route exact path="/characters">
                        Characters
                    </Route>
                    <Route exact path="/characters/:id">
                        Characters
                    </Route>
                    <Route path="/">
                        Home
                    </Route>
                </Switch>
            </main>
        </div>
    )


}