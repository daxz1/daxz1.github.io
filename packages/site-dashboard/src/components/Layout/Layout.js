import React from "react";

/**
 *
 * @param children
 * @returns {JSX.Element}
 * @constructor
 */
export const Layout = ({ children }) => (
  <div className="wrapper bg-colour bg-colour--woodsmoke">
    <div className="grid">{children}</div>
  </div>
);
