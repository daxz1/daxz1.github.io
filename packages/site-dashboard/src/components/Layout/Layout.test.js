/*global describe, test */

import React from "react";
import { create } from "react-test-renderer";
import { Layout } from "./Layout";

describe("Layout Component", () => {
  test("Matches the snapshot", () => {
    const Application = create(
      <Layout>
        <div>Layout</div>
      </Layout>
    );
    expect(Application.toJSON()).toMatchSnapshot();
  });
});
