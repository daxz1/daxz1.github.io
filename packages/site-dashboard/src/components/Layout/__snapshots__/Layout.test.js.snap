// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component Matches the snapshot 1`] = `
<div
  className="wrapper bg-colour bg-colour--woodsmoke"
>
  <div
    className="grid"
  >
    <div>
      Layout
    </div>
  </div>
</div>
`;
